lemur/certificates/cli.py

```diff
@@ -285,6 +285,171 @@ def rotate(endpoint_name, new_certificate_name, old_certificate_name, message, c
+)
+@manager.option(
+    "-a",
+    "--notify",
+    dest="message",
+    action="store_true",
+    help="Send a rotation notification to the certificates owner.",
+)
+@manager.option(
+    "-c",
+    "--commit",
+    dest="commit",
+    action="store_true",
+    default=False,
+    help="Persist changes.",
+)
+@manager.option(
+    "-r",
+    "--region",
+    dest="region",
+    action="store_true",
+    required=True,
+    help="Region in which to rotate the endpoint.",
+)
+def rotate_region(endpoint_name, new_certificate_name, old_certificate_name, message, commit, region):
+    """
+    Rotates an endpoint in a defined region it if it has not already been replaced. If it has
+    been replaced, will use the replacement certificate for the rotation.
+    :param old_certificate_name: Name of the certificate you wish to rotate.
+    :param new_certificate_name: Name of the certificate you wish to rotate to.
+    :param endpoint_name: Name of the endpoint you wish to rotate.
+    :param message: Send a rotation notification to the certificates owner.
+    :param commit: Persist changes.
+    :param region: Region in which to rotate the endpoint.
+    """
+    if commit:
+        print("[!] Running in COMMIT mode.")
+
+    print("[+] Starting endpoint rotation.")
+    status = FAILURE_METRIC_STATUS
+
+    try:
+        old_cert = validate_certificate(old_certificate_name)
+        new_cert = validate_certificate(new_certificate_name)
+        endpoint = validate_endpoint(endpoint_name)
+
+        if endpoint and new_cert:
+            if region in endpoint.dnsname:
+                log_data = f"[+] Rotating endpoint in region: {endpoint.dnsname} to certificate {new_cert.name}"
+                print(log_data)
+                current_app.logger.info(log_data)
+                request_rotation(endpoint, new_cert, message, commit)
+            else:
+                log_data = f"[+] Skipping rotation of {endpoint.dnsname} since not in {region}"
+                print(log_data)
+                current_app.logger.info(log_data)
+
+        elif old_cert and new_cert:
+            log_data = f"[+] Rotating all endpoints in {region} from {old_cert.name} to {new_cert.name}"
+            print(log_data)
+            current_app.logger.info(log_data)
+            for endpoint in old_cert.endpoints:
+                if region in endpoint.dnsname:
+                    log_data = f"[+] Rotating {endpoint.dnsname} in {region}"
+                    print(log_data)
+                    current_app.logger.info(log_data)
+                    request_rotation(endpoint, new_cert, message, commit)
+                else:
+                    log_data = f"[+] Skipping rotation of {endpoint.dnsname} since not in {region}"
+                    print(log_data)
+                    current_app.logger.info(log_data)
+        else:
+            log_data = f"[+] Rotating all endpoints that have new certificates available in {region}"
```

Comment:
It is not obvious from the docstring that this method will rotate all certificates in a region, which could have some negative consequences.  You could just update the docstring, or alternatively, break rotate_region() into two separate methods, something like rotate_region() to rotate an entire region and an updated version of rotate_certificate() that takes a region argument to rotate a single certificate.

Reply:
Looking at rotate(), the fact that it will rotate all certificates that are pending is not documented either.  May be worth just updating the docstrings and calling it a day.